src/sparseml/transformers/sparsification/obcq/obcq.py

```diff
@@ -90,25 +88,29 @@ def one_shot(
     config = AutoConfig.from_pretrained(model_path)
     model_type = config.model_type.lower()
 
-    model_loader_fn = None
-    forward_fn = None
     if "opt" in model_type:
-        model_loader_fn = SparseCausalLM.opt_model_from_pretrained
         forward_fn = opt_forward
     elif "llama" in model_type or "mistral" in model_type:
-        model_loader_fn = SparseCausalLM.auto_model_from_pretrained
         forward_fn = llama_forward
     else:
         _LOGGER.warning(
             f"A supported model type({SUPPORTED_MODELS}) could not be "
             f"parsed from model_path={model_path}. Defaulting to "
-            "AutoModelForCausalLM loading. "
+            "SparseAutoModel loading. "
         )
-        model_loader_fn = SparseCausalLM.auto_model_from_pretrained
         forward_fn = llama_forward
+
     torch_dtype = parse_dtype(precision)
-    model = model_loader_fn(
-        model_path, sequence_length=sequence_length, torch_dtype=torch_dtype
+    # create session and initialize a sparse model
+    session_manager.create_session()
+    model = initialize_sparse_model(
+        model_path=model_path,
+        task="text-generation",
+        sequence_length=sequence_length,
+        torch_dtype=torch_dtype,
+        config=config,
+        recipe=recipe_file,
```

Comment:
I think this is the reason for the failing tests. We're initializing the model with the recipe we want to apply to it, but it should be initialized with `input_recipe_path` instead, defined on deleted lines 130-137. `recipe_file` shouldn't get applied to the model until we call the `apply` function